### PR TITLE
[improvement] Query Panel - new design

### LINKED ISSUE #1949: Redesign Query panel (Editor)
### Query panel : new UI
- Adding new queries
- Selecting queries from the list
- New search box
- new UI for the REST API, Airtable and Stripe queries
- Minor: update transformation toggle to the primary colour, update information alert box (transformations)

### Screenshots (New UI)

6 search query panel
4 on hover query area
3 expanding the view
1 creating query
5 select the query section
Airtable
Stripe
Update record on airtable

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #1949 

Todo: 
- [x] Handle responsiveness related to query pane 
- [x] Add images/svgs for the selection component list
- [x] Handle/Fix dark theme for selection component
- [x] Update all other datasources' queries UI (should be consistent with Airtable query UI)
- [x] Fix minor bugs

### CODE REVIEW

File: frontend/src/Editor/QueryManager/QueryEditors/Restapi/TabBody.jsx [hunk 1]
```diff
@@ -1,58 +1,74 @@
...
+                  width="80%"
                   placeholder="value"
                   onChange={onChange('body', 1, index)}
                   componentName={`${componentName}/body::value::${index}`}
                 />
               </div>
               {index > 0 && (
                 <span
-                  className="btn-sm col-2 my-2"
+                  style={{ marginLeft: '-5%' }}
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/QueryManager/QueryEditors/Restapi/TabBody.jsx:64]: let's move static styles to `theme.scss`

File: frontend/src/Editor/Editor.jsx [hunk 2]
```diff
@@ -697,7 +704,8 @@ class Editor extends React.Component {
             </div>
           ) : (
             <button
-              className="btn badge bg-azure-lt"
+              style={{ marginTop: '3px' }}
+              className="btn badge bg-light-1"
```
Comment (gondar00) [inline frontend/src/Editor/Editor.jsx:708]:
```suggestion
              className="mt-3 btn badge bg-light-1"
```

File: frontend/src/Editor/QueryManager/QueryEditors/Restapi/index.jsx [hunk 3]
```diff
@@ -78,54 +78,58 @@ class Restapi extends React.Component {
...
-                  onChange={(value) => {
-                    changeOption(this, 'url', value);
-                  }}
-                  placeholder="Enter request URL"
-                  componentName={`${queryName}::url`}
-                />
-              </div>
+        <div className="row mt-2" style={{ height: 'fit-content' }}>
+          <div className="col-auto rest-methods-options" style={{ width: '90px' }}>
+            <SelectSearch
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryEditors/Restapi/index.jsx:83]: Better to use `react-select` here

File: frontend/src/Editor/QueryManager/QueryManager.jsx [hunk 4]
```diff
@@ -1,15 +1,17 @@
 import React from 'react';
 import { dataqueryService } from '@/_services';
 import { toast } from 'react-hot-toast';
-import Select from 'react-select';
+import SelectSearch, { fuzzySearch } from 'react-select-search';
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryManager.jsx:4]: Its better to use `react-select` here

File: frontend/src/Editor/QueryManager/QueryManager.jsx [hunk 5]
```diff
@@ -418,22 +369,26 @@ let QueryManager = class QueryManager extends React.Component {
...
                         }),
                       ]}
-                      onChange={(newValue) => this.changeDataSource(newValue.value)}
+                      value={selectedDataSource ? selectedDataSource.id : ''}
+                      search={true}
+                      onChange={(value) => this.changeDataSource(value)}
+                      filterOptions={fuzzySearch}
+                      renderOption={this.renderDataSourceOption}
                       placeholder="Select a data source"
-                      styles={selectStyles}
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryManager.jsx:436]: I thikn `selectStyles` is required for some dark mode fixes

File: frontend/src/_ui/Search/Search.jsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    };
+  }, []);
+
+  React.useEffect(() => {
+    return () => {
+      if (componentWillUnmount.current) {
+        setSearchText('');
+      }
+    };
+  }, [searchText]);
```
Comment (gondar00) [inline frontend/src/_ui/Search/Search.jsx:44]: If component unmounts and mounts again the `searchText` would be empty I assume, why is this required?

**Review by gondar00 — CHANGES_REQUESTED** [review]

File: frontend/src/Editor/QueryManager/QueryManager.jsx (hunk 5 above)
Comment (arpitnath) [reply frontend/src/Editor/QueryManager/QueryManager.jsx:436]: okay, for `react-select` we need selectStyles. resolving this

File: frontend/src/_ui/Search/Search.jsx (hunk 6 above)
Comment (arpitnath) [reply frontend/src/_ui/Search/Search.jsx:44]: similar bug to modal. but fixed that yesterday so we don't need that side effect to clear the fields on mounting and unmounting too

**Review by gondar00 — APPROVED** [review]

**Review by Navaneeth-pk — APPROVED** [review]
